codebuild/create_project.py

```diff
@@ -0,0 +1,153 @@
+        logging.debug(f'raw_env is {raw_env}')
+        env = raw_env.split(' ')
+    except AttributeError:
+        env = config.get(section, 'env').split(' ')
+        logging.debug(f'Section is {section}')
+
+    for i in env:
+        k, v = i.split("=")
+        env_list.append({"Name": k, "Value": v})
+
+    environment = Environment(
+        ComputeType=config.get(section, 'compute_type'),
+        Image=str(config.get(section, 'image')),
+        Type=str(config.get(section, 'env_type')),
+        PrivilegedMode=True,
+        EnvironmentVariables=env_list,
+    )
+
+    source = Source(
+        Location=config.get(section, 'source_location'),
+        Type=config.get(section, 'source_type'),
+        GitCloneDepth=config.get(section, 'source_clonedepth'),
+        BuildSpec=config.get(section, 'buildspec'),
+        ReportBuildStatus=True
+    )
+
+    project_id = project = Project(
+        project_name,
+        Artifacts=artifacts,
+        Environment=environment,
+        Name=project_name,
+        ServiceRole=Ref(service_role),
+        Source=source,
+        SourceVersion=config.get(section, 'source_version'),
+        BadgeEnabled=True,
+        DependsOn=service_role,
+    )
+    template.add_resource(project)
+    template.add_output([Output(f"CodeBuildProject{project_name}", Value=Ref(project_id))])
+
+def build_role(template=Template(), section="CFNRole", project_name:str=None, **kwargs) -> Ref:
+    """ Build a role with a CodeBuild managed policy. """
+    template.set_version('2010-09-09')
+    assert project_name
+    project_name+='Role'
+
+    # NOTE: By default CodeBuild manages the policies for this role.  If you delete a CFN stack and try to recreate the project
+    # or make changes to it when the Codebuild managed Policy still exists, you'll see an error in the UI:
+    # `The policy is attached to 0 entities but it must be attached to a single role`. (CFN fails with fail to update)
+    # Orphaned policies created by CodeBuild will have CodeBuildBasePolicy prepended to them; search for policies with this
+    # name and no role and delete to clear the error.
+    # TODO: Get a CloudFormation feature request to turn this off for project creation- let CFN manage the policy.
+    role_id = template.add_resource(
+        Role(
+            project_name,
+            Path='/',
+            AssumeRolePolicyDocument=PolicyDocument(
+                Statement=[
+                    Statement(
+                        Effect=Allow,
+                        Action=[AssumeRole],
+                        Principal=Principal("Service", ["codebuild.amazonaws.com"])
+                    )
+                ]
+            )
+        )
+    )
+
+    template.add_output([Output(project_name, Value=Ref(role_id))])
+    return Ref(role_id)
+
+def build_s3_cache(template=Template(), section=None, **kwargs) -> Template:
+    """ Create/Manage the s3 bucket for use by CodeBuild Cache    """
+    # TODO: Add s3 bucket.
+    pass
```

Comment:
Wanted to double check this is still an intentional no op.

Reply:
Unused, will remove in next revision.